src/controller/artifact/controller.go

```diff
@@ -128,28 +134,43 @@ func NewController() Controller {
 		regCli:       registry.Cli,
 		abstractor:   NewAbstractor(),
 		accessoryMgr: accessory.Mgr,
+		onDeleteFuncs: map[string]OnDeleteFunc{
```

Comment:
Should we handle these function into the handler of the delete artifact event
https://github.com/goharbor/harbor/blob/d03f0dcf2d6a420e66b486988c3dd00feaeb02d1/src/controller/event/handler/internal/artifact.go#L64 

current implementation seems create unnecessary dependency between artifact and job exection/ scan report

Reply:
good point.